torch/csrc/api/src/data/samplers/stream.cpp

```diff
@@ -20,7 +20,10 @@ BatchSize::operator size_t() const noexcept {
 
 StreamSampler::StreamSampler(size_t epoch_size) : epoch_size_(epoch_size) {}
 
-void StreamSampler::reset() {
+void StreamSampler::reset(optional<size_t> new_size) {
+  if (new_size.has_value()) {
+    epoch_size_ = new_size.value();
```

Comment:
We can use the faster, unchecked dereference operator `*new_size` here instead of another call to `.value()` , which is checked (because we checked that it has a value above)